src/main/java/org/mdbenefits/app/submission/actions/PreCheckPrograms.java

```diff
@@ -0,0 +1,39 @@
+package org.mdbenefits.app.submission.actions;
+
+import formflow.library.config.submission.Action;
+import formflow.library.data.Submission;
+import java.util.ArrayList;
+import lombok.extern.slf4j.Slf4j;
+import org.mdbenefits.app.data.enums.HelpNeededType;
+import org.mdbenefits.app.data.enums.ProgramType;
+import org.springframework.stereotype.Component;
+
+/**
+ *
+ */
+@Component
+@Slf4j
+public class PreCheckPrograms implements Action {
+
+    @Override
+    public void run(Submission submission) {
+        ArrayList<String> helpNeeded = (ArrayList<String>) submission.getInputData().get("helpNeeded[]");
+        ArrayList<String> programsToAdd = new ArrayList<>();
+
+        if (helpNeeded.contains("None")) {
+            submission.getInputData().remove("programs[]");
+        } else {
+            helpNeeded.forEach((String need) -> {
+                ArrayList<ProgramType> relevantPrograms = (ArrayList<ProgramType>) HelpNeededType.getRelevantProgramsFromValue(need);
```

Comment:
potentialPrograms could be a utility method that returns a flattened list of the HelpNeededType programTypes. We can use this flattened list to iterate on the choose Programs page.

This action stays as a way to set the checks for certain programs.